crates/compiler/builtins/roc/Num.roc

```diff
@@ -481,21 +481,50 @@ F32 : Num (FloatingPoint Binary32)
-## convert any [U64] to a [Dec] without losing information.
+## decimal places? It's the highest number of decimal places where you can
+## still convert any [U64] to a [Dec] without losing information.
 ##
 ## There are some use cases where [F64] and [F32] can be better choices than [Dec]
-## despite their precision issues. For example, in graphical applications they
-## can be a better choice for representing coordinates because they take up
+## despite their issues with base-10 numbers. For example, in graphical applications
+## they can be a better choice for representing coordinates because they take up
 ## less memory, certain relevant calculations run faster (see performance
-## details, below), and decimal precision loss isn't as big a concern when
+## details, below), and base-10 generally isn't as big a concern when
 ## dealing with screen coordinates as it is when dealing with currency.
 ##
+## Another scenario where [F64] can be a better choice than [Dec] is when representing
+## extremely small numbers. The smallest positive [F64] that can be represented without precision
+## loss is 2^(−1074), which is about 5 * 10^(-324). Here is that number next to the smallest
+## [Dec] that can be represented:
+##
+## * Smallest [F64]: 0.000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000005
+## * Smallest [Dec]: 0.000000000000000001
+##
+## This is because [floating-point](https://en.wikipedia.org/wiki/Floating-point_arithmetic) numbers
+## like [F64] can gain more digits of precision after the `.` when they aren't using as many digits
+## before the `.` - and this applies the most if the digit before `.` is `0`. So whereas [Dec] always
+## has 18 digits after the `.`, the number of digits after the `.` that [F32] can [F64] can represent
+## without precision loss depends on what comes before it.
+##
 ## ## Performance Details
 ##
-## [Dec] typically takes slightly less time than [F64] to perform addition and
-## subtraction, but 10-20 times longer to perform multiplication and division.
-## [sqrt] and trigonometry are massively slower with [Dec] than with [F64].
+## CPUs have dedicated instructions for many [F32] and [F64] operations, but none for [Dec].
+## Internally, [Dec] is represented as a 128-bit integer and uses multiple instructions to
+## perform fractional operations. This gives [F32] and [F64] performance advantages
+## for many operations.
+##
+## Here's a comparison of about how long [Dec] takes to perform a given operation compared to [F64],
+## based on benchmarks on an [M1](https://en.wikipedia.org/wiki/Apple_M1) CPU:
+## * [add] 0.75x
+## * [sub] 0.75x
+## * [mul] 4x
+## * [div] 32x
+## * [sin] 3x
+## * [asin] 9x
```

Comment:
btw @bhansconnect if you have a chance to benchmark the new additions, would be great to add them here for future reference!

Reply:
Do you want benchmarks for each function or just general categories? For example, I would assume sin, cos, and tan would be roughly similar.